[Cheong Ying Yi Clara] iP

claracheong4 opened the pull request.

Commit:
Add Gradle support

Commit:
Add Greet, Echo, Exit

Commit:
Add Add Task and List

Commit:
Add Mark Task as Done

Commit:
Add ToDos, Events, Deadlines

Commit:
Add Automated Text UI Testing

Commit:
Add Error Handling

Commit:
Make InvalidTaskNumberException's empty constructor more general

Commit:
Add Delete Task

Commit:
Add Enum TaskType and use it in related classes

Commit:
Add test cases for done and delete

Commit:
Add Save

Commit:
Add Date and Times

Commit:
Merge branch 'branch-Level-7' into master

* branch-Level-7:
  Add Save
  Add test cases for done and delete

Commit:
Merge branch 'branch-level-8' into master

* branch-level-8:
  Add Date and Times

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Task.java
#	src/main/java/Tasks.java
#	text-ui-test/input.txt

Commit:
More OOP: Add Parser, Command and related files, rename PrintDuke to Ui

Commit:
Organise java files into packages

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support' into master

* upstream/add-gradle-support:
  Add Gradle support

Commit:
Update Event.java

Commit:
Update Ui.java and EXPECTED.txt

Commit:
Add JUnit tests for duke.test package

Commit:
Update Task.java and runtest.bat: change status icon to unicode

Commit:
Update build.gradle and Add gradle related files

Commit:
Add Javadoc comments

Commit:
Update files to follow coding standard

Commit:
Add Find

Commit:
Merge branch 'branch-A-JavaDoc' into master

* branch-A-JavaDoc:
  Add Javadoc comments

Commit:
Merge branch 'branch-A-CodingStandard' into master

* branch-A-CodingStandard:
  Update files to follow coding standard

Commit:
Merge branch 'branch-Level-9' into master

Commit:
Update .gitignore and delete data files from testing

Commit:
Upgrade checkstyle

Commit:
Update javadocs for Ui.java

Commit:
build.gradle: Update version to 8.29

Commit:
Remove gradle setting files

Commit:
Add pmd

Commit:
Cleanup code

Add missing @Override

Commit:
Add GUI
Move Ui to ui package

Commit:
Add Customisation to GUI and messages

Commit:
Rename DukeException to BlueException

peter-yeh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,272 @@
+     * @throws EmptyInputException   If the task number is not in input.
+     * @throws UnknownInputException If the task number cannot be parsed.
+     */
+    private static DoneCommand getDoneCommand(String input) throws EmptyInputException,
```
Maybe this can be placed inside DoneCommand?

claracheong4 replied:
Sorry but don't know what 'this' refers to

peter-yeh commented on src/main/java/duke/exception/EmptyInputException.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * The Empty input exception.
+ */
+public class EmptyInputException extends DukeException {
```
It's nice you separated out this exception instead of also classifying empty input as a DukeException

peter-yeh commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,77 @@
+ * The Event, a task that starts at a specific time and ends at a specific time.
+ */
+public class Event extends Task {
+    /**
+     * The Date.
+     */
+    private final LocalDate date;
```
Is this a bit unnecessary because the variable date is self explanatory

peter-yeh commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,25 @@
+package duke.task;
+
+/**
+ * The Todo, a task without any date or time attached to it.
+ */
```
Maybe the head comments can be shrunk to one line?
This would make the code look more compact and neater if the header comment is not too long.
I noticed there were some quite short comments which could have been shorten to one line
`` /** The Todo, ... to it. */ ``

claracheong4 replied:
Hmm I think all the comments in relation to javadocs is bcs I autogenerated javadocs for all elements and just edited it afterwards. I personally think that this looks nicer when I open the javadocs .html pages, maybe not in the .java files.

peter-yeh commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,49 @@
+todo
+todo 
+todo dfsfsdf
+event
+event 
```
It is good that you have so many test cases, this shows that you test your codes thoroughly :)

peter-yeh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,184 @@
+ * The Ui interacts with the user..
+ */
+public class Ui {
+    /**
+     * The Scanner to read inputs.
+     */
+    private final Scanner scanner;
```
Maybe this code is self-explanatory? There were also many instances which I feel that the variable name is self-explanatory.

peter-yeh commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,34 @@
+ * The Bye command which terminates the program.
+ */
+public class ByeCommand extends Command {
+    /**
+     * The constant EXIT_STATUS.
+     */
```
Maybe this does not need a comment header?

peter-yeh reviewed (commented):
Hi, this was a really well written code! I cannot find any major error after spending more than 15 mins looking through. So I can only comment on those very minor java docs stuff. Cheers :)

Commit:
Update Customisation

Move inline css in fxml files to base.css
Add random generation of bye exit message

ChenXJ98 commented on src/main/java/duke/command/CommandType.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * The enum Command type.
+ */
+public enum CommandType {
+    /**
+     * Bye command type.
+     */
+    BYE,
+    /**
+     * List command type.
+     */
+    LIST,
+    /**
+     * Delete command type.
+     */
+    DELETE,
+    /**
+     * Done command type.
+     */
+    DONE,
+    /**
+     * Task command type.
+     */
+    TASK,
+    /**
+     * Find command type.
+     */
+    FIND
+}
```
Might not be necessary to indicate all the command types in javadocs.

claracheong4 replied:
Hmm I think all the comments in relation to javadocs is bcs I autogenerated javadocs for all elements and just edited it afterwards. I personally think that this looks nicer when I open the javadocs .html pages, maybe not in the .java files.

ChenXJ98 reviewed (commented):
Looks really good overall, just some nit picks!

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support' into master

* upstream/add-gradle-support:
  build.gradle: Update version to 8.29

Commit:
Merge branch 'branch-Level-10' into master

* branch-Level-10:
  Update Customisation
  Rename DukeException to BlueException
  Add Customisation to GUI and messages
  Add GUI Move Ui to ui package
  Cleanup code
  Add pmd
  Remove gradle setting files

Commit:
Update build.gradle

Commit:
Update message for list command

Commit:
Update MainWindow.fxml

Commit:
Add Assertions

Commit:
Improve code quality

claracheong4 committed:
Merge pull request #1 from claracheong4/branch-A-Assertions

Add Assertions

claracheong4 committed:
Merge pull request #2 from claracheong4/branch-A-CodeQuality

Improve code quality

Commit:
Add Start and End Time to Events and Add reminder for today's tasks

Commit:
Merge branch 'branch-BCD-Extension' into master

Commit:
Add FindTodayCommand

Use FindTodayCommand instead of FindDateCommand instead of today

Commit:
Adjust positioning

Commit:
Add README and Ui (screenshot)

Commit:
Update README in root folder

Commit:
Update README

Reorder contents

claracheong4 committed:
Set theme jekyll-theme-architect

claracheong4 committed:
Set theme jekyll-theme-slate

claracheong4 committed:
Set theme jekyll-theme-slate

claracheong4 committed:
Set theme jekyll-theme-slate

claracheong4 committed:
Update README.md

claracheong4 committed:
Update README.md

claracheong4 committed:
Update README.md

claracheong4 committed:
Update README.md
